network.go

```diff
@@ -944,7 +945,10 @@ func (n *network) ipamAllocateVersion(ipVer int, ipam ipamapi.Ipam) error {
 		// irrespective of whether ipam driver returned a gateway already.
 		// If none of the above is true, libnetwork will allocate one.
 		if cfg.Gateway != "" || d.Gateway == nil {
-			if d.Gateway, _, err = ipam.RequestAddress(d.PoolID, net.ParseIP(cfg.Gateway), nil); err != nil {
+			var gatewayOpts = map[string]string{
+				netlabel.Gateway: "",
```

Comment:
@mavenugo Go is not my native language so would this be the correct way to do it?

ipamapi/contract.go

In the const declaration add

```
// RequestAddressType represents the Address Type used when requesting an address
RequestAddressType = "RequestAddressType"
```

and in network.go

```
var gatewayOpts = map[string]string{
    ipamapi.RequestAddressType: netlabel.Gateway,
}
```